### ISSUE TITLE
Template relation doesn't appear on graph

### ISSUE THREAD

**Issue by Sblawy** [doxygen/doxygen#11144]
**Describe the bug**
While using doxygen to generate relation graph, the templated element appear but the link (orange arrow) with the base template is  not present.
In the next example, the 2 structs have the template but only one appear in graph.
The only one who appear is the one who extends the templated class.

**Screenshots**
<img width="366" alt="class_hierarchy" src="https://github.com/user-attachments/assets/def78d2d-362d-4474-a0f4-bfce6702fe1a">
<img width="572" alt="inherited_template_struct_doc" src="https://github.com/user-attachments/assets/4c6546d1-0e96-44e7-a43f-644ac74ea74c">
<img width="578" alt="template_struct_doc" src="https://github.com/user-attachments/assets/c9d9afa0-caad-40bf-a13f-8c3102667527">

**To Reproduce**
The Doxyfile is attached to the issue, and the code used in sceen is here :
```
namespace test_package {
  struct T_1;
  struct T_2;
  template <typename T> struct TemplaceStruct {};

  struct HeritedStruct1: public TemplaceStruct<T_1> {
    static int getValue()
    {
      return 1;
    }
  };

  struct TemplaceStruct<T_2> {
    static int getValue()
    {
      return 2;
    }
  };
}
```

**Expected behavior**
In this case, the orange arrow to identify the template need to appear for `struct TemplaceStruct<T_2>.`
Ideally, having a page with all relationships (inheritance and template) will be a most.

**Version**
1.12.0

**Comment by albert-github** [comment]
> The Doxyfile is attached to the issue

Sorry I don't see the `Doxyfile` please add it, actually better:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

**Comment by Sblawy** [comment]
Excuse me, I possibly do an error when I add the Doxyfile.
Here is attached a zip file with source code, Doxyfile and the documentation generate by doxygen.
[test_template_doxygen.zip](https://github.com/user-attachments/files/17025955/test_template_doxygen.zip)

**Comment by doxygen** [comment]
@Sblawy Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.13.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
